### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by petrGB**

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -29,16 +29,36 @@ func main() {
...
+	logCf := zap.Config{
+		Level:       zap.NewAtomicLevelAt(zapcore.Level(cfg.LogLevel)),
+		Development: false,
+		Sampling: &zap.SamplingConfig{
+			Initial:    100,
+			Thereafter: 100,
+		},
+		Encoding:         "console",
+		EncoderConfig:    zap.NewDevelopmentEncoderConfig(),
+		OutputPaths:      []string{"stderr"},
```
Comment (DmSide) [inline cmd/main.go:41]: stdout

File: cmd/main.go [hunk 2]
```diff
@@ -73,18 +93,18 @@ func processResult(ctx context.Context, cancel func(), cr crawler.Crawler, cfg c
 		case msg := <-cr.ChanResult():
 			if msg.Err != nil {
 				maxErrors--
-				log.Printf("crawler result return err: %s\n", msg.Err.Error())
+				log.Info("crawler result", zap.String("err", msg.Err.Error()))
```
Comment (DmSide) [inline cmd/main.go:96]: maybe should be "Error" or "Warn" level

File: internal/crawler/crawler.go [hunk 3]
```diff
@@ -63,6 +68,9 @@ func (c *crawler) DepthDiff(diff int32) {
 func (c *crawler) Scan(ctx context.Context, url string, depth int) {
 	defer c.wg.Done()
 
+	c.log.Debug("start", zap.String("url", url), zap.Int("url", depth))
```
Comment (DmSide) [inline internal/crawler/crawler.go:71]: zap.Int("url", depth)
to
zap.Int("depth", depth)

**Comment by t0pep0** [comment]
Принято

**Pull request closed without merging**